src/DynamoCore/Visualization/RenderPackageCache.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using Autodesk.DesignScript.Interfaces;
+
+namespace Dynamo.Visualization
+{
+    /// <summary>
+    /// This class controls the collection and distribution of render packages
+    /// </summary>
+    public class RenderPackageCache
+    {
+        private List<IRenderPackage> packages;
+        private Dictionary<Guid, RenderPackageCache> portMap;
+
+        private void AddPort(IRenderPackage package, Guid outputPortId)
+        {
+            if (portMap == null)
+            {
+                portMap = new Dictionary<Guid, RenderPackageCache>();
+            }
+
+            if (!portMap.ContainsKey(outputPortId))
+            {
+                portMap[outputPortId] = new RenderPackageCache();
+            }
+
+            portMap[outputPortId].Add(package);
+        }
+
+        /// <summary>
+        /// Create an empty RenderPackageCache object
+        /// </summary>
+        public RenderPackageCache()
+        {
+            packages = new List<IRenderPackage>();
+            portMap = null;
+        }
+
+        /// <summary>
+        /// Create a RenderPackageCache object containing the given render packages
+        /// </summary>
+        /// <param name="otherPackages">The packages to fill the new cache object.</param>
+        public RenderPackageCache(IEnumerable<IRenderPackage> otherPackages)
+        : this()
+        {
+            packages.AddRange(otherPackages);
+        }
+
+        /// <summary>
+        /// Get the render packages in this cache
+        /// </summary>
+        public IEnumerable<IRenderPackage> Packages
+        {
+            get
+            {
+                return packages;
+            }
+        }
+
+        /// <summary>
+        /// Get the RenderPackageCache object for the given port ID
+        /// </summary>
+        /// <param name="portId">The port ID used to get the sub-cache.</param>
+        public RenderPackageCache GetPortPackages(Guid portId)
+        {
+            if (portMap == null)
```

Comment:
please use braces or change to one line:
https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards#coding-standards

Reply:
I will update this to use braces. 

On a couple of side notes (not trying to start the brace wars here...);
1) For a single line after an if I have not heard a convincing argument why there should be braces around the single line. IMHO this should not be in the coding standards. YMMV, opinions (with data based arguments) solicited please.
2) Having both the if and the return line on one line is a bad idea. It makes debugging much less clear, as to what would be broken on by a breakpoint; is it the if or the resulting statement if the if is true? I think this should not be allowed in the standards. If it is, why should there not be braces around the statement in the if even if the statement is on the same line as the if? The only difference is whitespace.